File: Packages/com.bci-runtime.shm/Samples~/BCIDashboard/ShmConfig.cs
using UnityEngine;
using System.IO;
using System.Text.RegularExpressions;

public static class ShmConfig
{
    private static bool _loaded;
    private static string _shmName = "Local_BCI_GustationRing";
    private static string _returnShmName = "Local_BCI_GustationRing_Return";
    private static int _ringBufferSize = 32;
    private static int _maxChannels = 8;
    private static int _headerSize = 24;
    private static int _returnRingSize = 8;

    public static string ShmName { get { Load(); return _shmName; } }
    public static string ReturnShmName { get { Load(); return _returnShmName; } }
    public static int RingBufferSize { get { Load(); return _ringBufferSize; } }
    public static int MaxChannels { get { Load(); return _maxChannels; } }
    public static int HeaderSize { get { Load(); return _headerSize; } }
    public static int ReturnRingSize { get { Load(); return _returnRingSize; } }

    static void Load()
    {
        if (_loaded) return;
        _loaded = true;

        string path = Path.Combine(Application.streamingAssetsPath, "shm_config.json");
        if (!File.Exists(path)) return;

        try
        {
            string json = File.ReadAllText(path);
            var m = Regex.Match(json, "\"shm_name\"\\s*:\\s*\"([^\"]+)\"");
            if (m.Success) _shmName = m.Groups[1].Value;
            m = Regex.Match(json, "\"return_shm_name\"\\s*:\\s*\"([^\"]+)\"");
            if (m.Success) _returnShmName = m.Groups[1].Value;
            m = Regex.Match(json, "\"ring_buffer_size\"\\s*:\\s*(\\d+)");
            if (m.Success) _ringBufferSize = int.Parse(m.Groups[1].Value);
            m = Regex.Match(json, "\"max_channels\"\\s*:\\s*(\\d+)");
            if (m.Success) _maxChannels = int.Parse(m.Groups[1].Value);
            m = Regex.Match(json, "\"header_size\"\\s*:\\s*(\\d+)");
            if (m.Success) _headerSize = int.Parse(m.Groups[1].Value);
            m = Regex.Match(json, "\"return_ring_buffer_size\"\\s*:\\s*(\\d+)");
            if (m.Success) _returnRingSize = int.Parse(m.Groups[1].Value);
        }
        catch { }
    }
}

File: core/shm_config.py
"""Load SHM configuration from shm_config.json.

Shared between Python tools and C# runtime (via Resources).
Looks for the config file relative to the project root or the calling script.
"""
import json, os
from pathlib import Path

DEFAULT_CONFIG = {
    "shm_name": "Local_BCI_GustationRing",
    "return_shm_name": "Local_BCI_GustationRing_Return",
    "ring_buffer_size": 32,
    "slot_header_size": 8,
    "channel_header_size": 8,
    "channel_stride": 16,
    "max_channels": 8,
    "header_size": 24,
    "poll_interval_ms": 4,
    "heartbeat_timeout_sec": 3.0,
    "return_ring_buffer_size": 8,
}

_config = None


def load_config(path=None):
    global _config
    if _config is not None:
        return _config

    if path is None:
        candidates = [
            Path(os.getcwd()) / "shm_config.json",
            Path(__file__).parent.parent / "shm_config.json",
            Path(__file__).parent.parent.parent / "shm_config.json",
        ]
        for c in candidates:
            if c.exists():
                path = str(c)
                break

    if path and os.path.exists(path):
        with open(path) as f:
            cfg = json.load(f)
            _config = {**DEFAULT_CONFIG, **cfg}
            return _config
    _config = dict(DEFAULT_CONFIG)
    return _config


def get(key, default=None):
    cfg = load_config()
    return cfg.get(key, default)
